src/witan/app/handler.clj

```diff
@@ -27,24 +31,21 @@
 
 (defn login
   [body]
-  (let [username (:username body)
-        password (:password body)
-        valid? (user/user-valid? username password)]
-    (if valid?
+  (let [{:keys [username password]} body
```

Comment:
You could avoid this `let` by destructuring in the fn definition: `{:keys [username password] :as body}`